lib/validators/specValidator.js

```diff
@@ -74,6 +76,12 @@ class SpecValidator {
     if (options.shouldResolvePureObjects === null || options.shouldResolvePureObjects === undefined) {
       options.shouldResolvePureObjects = true;
     }
+    if (options.shouldResolveDiscriminator === null || options.shouldResolveDiscriminator === undefined) {
```

Comment:
if shouldResolveDiscriminator is passed as "true" in options, should we also make sure to set "shouldResolveAllOf" as true?

Reply:
Yup. I am doing that below in lines 81 and 82

Reply:
Yes, but isn't that happening only if "shouldResolveDiscriminator" is null or undefined? how about if someone set it to true?

Reply:
Got it. Have changed it now. Please check the new commit.